backend/pkg/api/connect/integration/acl_test.go

```diff
@@ -0,0 +1,229 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"connectrpc.com/connect"
+	"github.com/carlmjohnson/requests"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/twmb/franz-go/pkg/kadm"
+
+	"github.com/redpanda-data/console/backend/pkg/protocmp"
+	v1alpha1 "github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/dataplane/v1alpha1"
+	v1alpha1connect "github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/dataplane/v1alpha1/dataplanev1alpha1connect"
+)
+
+func (s *APISuite) DeleteAllACLs(ctx context.Context) error {
+	acls := kadm.NewACLs().
+		Allow().
+		ResourcePatternType(kadm.ACLPatternAny).
+		Groups().
+		Topics().
+		TransactionalIDs().
+		Clusters().
+		Operations(kadm.OpAny)
+	if err := acls.ValidateDelete(); err != nil {
+		return fmt.Errorf("failed to validate delete acl request: %w", err)
+	}
+
+	_, err := s.kafkaAdminClient.DeleteACLs(ctx, acls)
+	return err
+}
+
+func (s *APISuite) newStdACLResource(resourceType v1alpha1.ACL_ResourceType, resourceName string, principal string) *v1alpha1.ListACLsResponse_Resource {
+	return &v1alpha1.ListACLsResponse_Resource{
+		ResourceType:        resourceType,
+		ResourceName:        resourceName,
+		ResourcePatternType: v1alpha1.ACL_RESOURCE_PATTERN_TYPE_LITERAL,
+		Acls: []*v1alpha1.ListACLsResponse_Policy{
+			{
+				Principal:      principal,
+				Host:           "*",
+				Operation:      v1alpha1.ACL_OPERATION_ALL,
+				PermissionType: v1alpha1.ACL_PERMISSION_TYPE_ALLOW,
+			},
+		},
+	}
+}
+
+func (s *APISuite) TestListACLs() {
+	t := s.T()
+	require := require.New(t)
+	assert := assert.New(t)
+
+	t.Run("list ACLs with default request (connect-go)", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		defer cancel()
+
+		// 1. Seed some ACLs
+		principal := "User:test"
+		resourceNamePrefix := "console-test-"
+
+		resourceNames := map[v1alpha1.ACL_ResourceType]string{
+			v1alpha1.ACL_RESOURCE_TYPE_GROUP:            fmt.Sprintf("%vgroup", resourceNamePrefix),
+			v1alpha1.ACL_RESOURCE_TYPE_TOPIC:            fmt.Sprintf("%vtopic", resourceNamePrefix),
+			v1alpha1.ACL_RESOURCE_TYPE_TRANSACTIONAL_ID: fmt.Sprintf("%vtransactional-id", resourceNamePrefix),
+			v1alpha1.ACL_RESOURCE_TYPE_CLUSTER:          "kafka-cluster",
```

Comment:
Minor nit or check, but this key doesn't seem to be used? Should it be used somewhere? or removed?

Reply:
Nvm we iterate over the map.